Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +10,232 @@ public static class BalanceStats
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+
+            var HighestBalanceEver = decimal.MinValue;
+            var PersonWithHighestBalance = new StringBuilder();
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                var highestAmountOfPerson = decimal.Parse(currentPersonData[1..].Max());
+
+                if (highestAmountOfPerson > HighestBalanceEver)
+                {
+                    HighestBalanceEver = highestAmountOfPerson;
+                    var currentOne = PersonWithHighestBalance.ToString();
+
+                    if (string.IsNullOrEmpty(currentOne))
+                    {
+                        PersonWithHighestBalance.Append(currentPersonData[0]);
```

Comment:
If you found a new max, I think that means a new name and a new message in all cases. So you might just reset the string builder and append the new name.

Reply:
so you mean instead of this :  ` personWithHighestBalance.Replace(currentOne, currentPersonData[0]);`   use ` PersonWithHighestBalance.Reset;  PersonWithHighestBalance.Apppend(new Person)  `  ?

Reply:
Yes

Reply:
oke, I will change it.